### PR TITLE
New sample: Display device location with NMEA data sources

### PR DESCRIPTION
**Pull request by ZackAllen**

### CODE REVIEW

File: src/WinUI/ArcGISRuntime.WinUI.Viewer/MainPage.xaml.cs [hunk 1]
```diff
@@ -81,12 +79,12 @@ private void LoadTreeView(SearchableTreeNode fullTree)
             }
             CategoriesTree.RootNodes.Clear();
 
-            muxc.TreeViewNode rootNode;
```
Comment (nCastle1) [inline src/WinUI/ArcGISRuntime.WinUI.Viewer/MainPage.xaml.cs:84]: Was this intended for this PR?

**Review by nCastle1 — COMMENTED** [review]
Sample looks good, I have some initial suggestions, but use your judgment.

File: src/WinUI/ArcGISRuntime.WinUI.Viewer/MainPage.xaml.cs (hunk 1 above)
Comment (ZackAllen) [reply src/WinUI/ArcGISRuntime.WinUI.Viewer/MainPage.xaml.cs:84]: This is some minor cleanup I did while editing this file. That import and prefix are not needed in WinUI, I believe they were in UWP.

File: src/Forms/Shared/Samples/Location/LocationWithNMEA/LocationWithNMEA.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            _nmeaSource = new NmeaLocationDataSource(SpatialReferences.Wgs84);
+
+            // Android and UWP:
+            // To create a NmeaLocationDataSource for a bluetooth device, use the `FromBluetooth` constructor. https://developers.arcgis.com/net/api-reference/api/android/Esri.ArcGISRuntime/Esri.ArcGISRuntime.Location.NmeaLocationDataSource.FromBluetooth.html
+            // To create a NmeaLocationDataSource from a serial port, use the `FromSerialPort` constructor. https://developers.arcgis.com/net/api-reference/api/android/Esri.ArcGISRuntime/Esri.ArcGISRuntime.Location.NmeaLocationDataSource.FromSerialPort.html
+
+            // iOS:
+            // When using an NMEA device on iOS, use the `NmeaLocationDataSource.FromAccessory` constructor. https://developers.arcgis.com/net/api-reference/api/ios/Esri.ArcGISRuntime/Esri.ArcGISRuntime.Location.NmeaLocationDataSource.FromAccessory.html
+
+            // Set the location data source to use the stream from our simulator.
```
Comment (nCastle1) [inline src/Forms/Shared/Samples/Location/LocationWithNMEA/LocationWithNMEA.xaml.cs:62]:
```suggestion
            // Set the location data source to use the stream from the simulator.
```

File: src/Forms/Shared/Samples/Location/LocationWithNMEA/readme.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+* com.amanenterprises.nmeasource
+* com.dualav.xgps150
+* com.garmin.pvt
+* com.junipersys.geode
+* com.leica-geosystems.zeno.gnss
+* com.searanllc.serial
+* com.trimble.correction, com.trimble.command (1)
+
+(1) Some Trimble models requires a proprietary SDK for NMEA output.
```
Comment (nCastle1) [inline src/Forms/Shared/Samples/Location/LocationWithNMEA/readme.md:57]:
```suggestion
(1) Some Trimble models require a proprietary SDK for NMEA output.
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Location/LocationWithNMEA/LocationWithNMEA.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            _lineCounter = 0;
+        }
+
+        private void TimerElapsed(object sender, ElapsedEventArgs e)
+        {
+            // Get the next NMEA string and append return and newline characters to it.
+            string nmeaString = $"{_nmeaStrings[_lineCounter]}\r\n";
+
+            // Check if the start of a new location message.
+            if (nmeaString.StartsWith("GPGGA"))
```
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Location/LocationWithNMEA/LocationWithNMEA.xaml.cs:187]: This isn't working, the messages start with `$`

**Review by nCastle1 — CHANGES_REQUESTED** [review]
I made a few suggestions, those suggestions apply to other platforms.

**Review by nCastle1 — CHANGES_REQUESTED** [review]
I'd like to see two improvements to make this easier to use:

- Speed up the interval or update the data so that changes happen more frequently. It looked like it was broken at first, but it was just taking a long time to update. Maybe, you could show the latest NMEA message. Maybe a speed increase would be enough.
- Add an improved help label indicating the status: not started, waiting for first message, etc.

Optional suggestions:

- Maybe Recenter should just be on by default? I don't know if the extra button is necessary, especially since the demo doesn't work very well without recenter on.
- Reset should clear all of the labels

**Review by nCastle1 — COMMENTED** [review]
I'm seeing strange behavior on Forms UWP. UI tends to resize a lot due to the length of the label changing. That can be fixed by setting `LineBreakMode` to `TailTruncation`. 

Rather than disabling panning, I recommend re-enabling autopan mode when the location updates. I thought something was broken at first.

**Review by nCastle1 — APPROVED** [review]

**Pull request merged**